shark-graph/src/test/java/shark/LongScatterSetTest.kt

```diff
@@ -0,0 +1,131 @@
+package shark
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Ignore
+import org.junit.Test
+import shark.internal.hppc.HHPC
+import shark.internal.hppc.LongScatterSet
+
+class LongScatterSetTest {
+
+  @Test fun `verify add operation`() {
+    val set = LongScatterSet()
+    verifyAddOperation(set)
+  }
+
+  @Test fun `verify remove operation`() {
+    val set = LongScatterSet()
+    verifyRemoveOperation(set)
+  }
+
+  @Test fun `verify release operation`() {
+    val set = LongScatterSet()
+    set.add(42)
+    set.release()
+
+    assertFalse(42 in set)
+    assertEquals(0, set.size())
+  }
+
+  @Test fun `verify ensureCapacity does not break operations`() {
+    // Ensure capacity before adding/removing elements
+    verifyAddOperation(LongScatterSet().apply { ensureCapacity(10) })
+    verifyRemoveOperation(LongScatterSet().apply { ensureCapacity(10) })
```

Comment:
This is the "arrange" part of the test. There shouldn't be any verification here, only set up. Asserts should stick to the last phase of a test.

Reply:
This is the actual testing part; but I can rewrite it to be cleaner